### PR TITLE
Add autogenes Python package

### PR DESCRIPTION
**Pull request by PelzKo**
Added the AutoGeneS Python package

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/autogenes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - python >=3.6
+    - pandas >=0.25.1
+    - anndata >=0.6.22.post1
+    - numpy >=1.17.2
+    - dill >=0.3.1.1
+    - deap >=1.3.0
+    - scipy >=1.3
+    - cachetools >=3.1.1
+    - scikit-learn >=0.21.3
+    - matplotlib >=3.0
```
Comment (grst) [inline recipes/autogenes/meta.yaml:28]: `noarch: python` packages don't require all dependencies under `host`. It's usually just `python` and `pip`. 

If a package has some `C` dependencies, you would additionally require a compiler. 

Explanation: `host` are the packages needed to *build/install* the package, `run` are the dependencies required to *run/execute* the package.

File: recipes/autogenes/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+source:
+  url:
+    - https://files.pythonhosted.org/packages/f0/04/4768c40962c38dd15b6cbd67d23f1c0d97d2dedeaa6558f6f8689458b4c5/autogenes-{{ version }}.tar.gz
+  sha256: 8daa6e9aa97c7ddf45ab0906307a4875a25bd99301be0838bb17e313d837446a
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
+  noarch: generic
```
Comment (grst) [inline recipes/autogenes/meta.yaml:15]: should be `noarch: python`

**Review by grst — CHANGES_REQUESTED** [review]

File: recipes/autogenes/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{% set version = "1.0.4" %}
+
+package:
+  name: autogenes
+  version: {{ version }}
+
+source:
+  url:
+    - https://files.pythonhosted.org/packages/f0/04/4768c40962c38dd15b6cbd67d23f1c0d97d2dedeaa6558f6f8689458b4c5/autogenes-{{ version }}.tar.gz
```
Comment (grst) [inline recipes/autogenes/meta.yaml:9]: I think this URL won't work once the version gets bumped... 
It usually looks more like this `https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz`

File: recipes/autogenes/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{% set version = "1.0.4" %}
+{% set name = "autogenes" %}
+
+package:
+  name: autogenes
+  version: {{ version }}
+
+source:
+  url:
+    - http://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (grst) [inline recipes/autogenes/meta.yaml:10]:
```suggestion
    - https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

**Comment by PelzKo** [comment]
@BiocondaBot please add label

**Review by grst — APPROVED** [review]
LGTM!

**Comment by druvus** [comment]
@BiocondaBot please merge

**Pull request merged**